# Applied Statistics Practical 10 part 2

ANOVA: Tests whether the means of two or more independent samples are significantly different.

Assumptions


Observations in each sample are independent and identically distributed (iid). Observations in each sample are normally distributed. Observations in each sample have the same variance. Interpretation

H0: the means of the samples are equal. H1: one or more of the means of the samples are unequal.

In [1]:
from scipy.stats import f_oneway
data1 = [0.873, 2.817, 0.121, -0.945, -0.055, -1.436, 0.360, -1.478, -1.637, -1.869]
data2 = [1.142, -0.432, -0.938, -0.729, -0.846, -0.157, 0.500, 1.183, -1.075, -0.169]
data3 = [-0.208, 0.696, 0.928, -1.148, -0.213, 0.229, 0.137, 0.269, -0.870, -1.204]
stat, p=f_oneway(data1,data2,data3)
print('stat=%.3f, p=%.3f' % (stat,p))
if p>0.05:
    print('Probably the same distribution')
else:
    print('Probably the same distribution')

stat=0.096, p=0.908
Probably the same distribution


An F-test is used to test whether two population variances are equal. The null and alternative hypotheses for the test are as follows:

H0: σ12 = σ22 (the population variances are equal)

H1: σ12 ≠ σ22 (the population variances are not equal)

In [2]:
x = [18, 19, 22, 25, 27, 28, 41, 45, 51, 55]
y = [14, 15, 15, 17, 18, 22, 25, 25, 27, 34]

In [3]:
import numpy as np
import scipy
def f_test(x,y):
    x=np.array(x)
    y=np.array(y)
    f=np.var(x,ddof=1)/np.var(y,ddof=1)
    dfn=x.size-1
    dfd=y.size-1
    p=1-scipy.stats.f.cdf(f,dfn,dfd)
    return f,p
f_test(x,y)

(4.387122002085505, 0.01912653593238578)

Kruskal-Wallis H-test for independent samples. The Kruskal-Wallis H-test tests the null hypothesis that the population median of all of the groups are equal. It is a non-parametric version of ANOVA. The test works on 2 or more independent samples, which may have different sizes. Note that rejecting the null hypothesis does not indicate which of the groups differs. Post hoc comparisons between groups are required to determine which groups are different.

In [4]:
from scipy import stats
x=[1,3,5,7,9]
y=[2,4,6,8,10]
print(stats.kruskal(x,y))

KruskalResult(statistic=0.2727272727272734, pvalue=0.6015081344405895)


p>0.05, fail to reject null, conclusion: population median of all groups will be same

In [5]:
x=[1,1,1]
y=[2,2,2]
z=[2,2]
stats.kruskal(x,y,z)

KruskalResult(statistic=7.0, pvalue=0.0301973834223185)

In [6]:
group1=[7,14,14,13,12,9,6,14,12,8]
group2=[15,17,13,15,15,13,9,12,10,8]
group3=[6,8,8,9,5,14,13,8,10,9]

In [7]:
print(stats.kruskal(group1,group2,group3))

KruskalResult(statistic=6.287801578353988, pvalue=0.043114289703508814)


p<0.05, reject null, conclusion: population median of all groups is not same but we do not know which group differ.

#### A Mann-Wehiteney
A Mann-Whitney U test
(sometimes called the Wilcoxon rank-sum test) is used to compare the differences between two samples when the sample distributions are not normally distributed and the sample sizes are small (n <30). It is considered to be the nonparametric equivalent to the two sample t-test. Researchers want to know if a fuel treatment leads to a change in the average mpg of a car. To test this, they measure the mpg of 12 cars with the fuel treatment and 12 cars without it. Since the sample sizes are small and the researchers suspect that the sample distributions are not normally distributed, they decided to perform a Mann-Whitney U test to determine if there is a statistically significant difference in mpg between the two groups

In [8]:
group1 = [20, 23, 21, 25, 18, 17, 18, 24, 20, 24, 23, 19]
group2 = [24, 25, 21, 22, 23, 18, 17, 28, 24, 27, 21, 23]

In [9]:
import scipy.stats as stats
stats.mannwhitneyu(group1,group2,alternative='two-sided')

MannwhitneyuResult(statistic=50.0, pvalue=0.21138945901258455)

The test statistic is 50.0 and the corresponding two-sided p-value is 0.2114.

In this example, the Mann-Whitney U Test uses the following null and alternative hypotheses:

H0: The mpg is equal between the two groups

HA: The mpg is not equal between the two groups

Since the p-value (0.2114) is not less than 0.05, we fail to reject the null hypothesis. We do not have sufficient evidence to say that the true mean mpg is different between the two groups.